server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/model/vote/IRVChoicesTests.java

```diff
@@ -0,0 +1,409 @@
+    testUtils.log(LOGGER,"overvotesAndSkipsTest1");
+
+    IRVChoices b = new IRVChoices("Alice(1),Bob(2),Chuan(2),Diego(4)");
+    assertEqualListsOfStrings(List.of("Alice"), b.getValidIntentAsOrderedList());
+  }
+
+  /**
+   * Apply rule 26.7.3 - removing all but the highest expressed preference for a given candidate.
+   * Result is a single valid 1st preference for Alice.
+   * @throws IRVParsingException never
+   */
+  @Test
+  public void duplicateCandidatesTest1() throws IRVParsingException {
+    testUtils.log(LOGGER,"duplicateCandidatesTest1");
+
+    IRVChoices b = new IRVChoices("Alice(2),Alice(1)");
+    assertEqualListsOfStrings(List.of("Alice"), b.getValidIntentAsOrderedList());
+  }
+
+  /**
+   * Apply rule 26.7.3 - removing all but the highest expressed preference for a given candidate.
+   * The second preference for Alice is removed, leaving two valid preferences.
+   * @throws IRVParsingException never
+   */
+  @Test
+  public void duplicateCandidatesTest2() throws IRVParsingException {
+    testUtils.log(LOGGER,"duplicateCandidatesTest2");
+
+    IRVChoices b = new IRVChoices("Alice(1),Alice(2),Bob(2)");
+    assertEqualListsOfStrings(List.of("Alice","Bob"), b.getValidIntentAsOrderedList());
+  }
+
+  /**
+   * Apply rule 26.7.3 - removing all but the highest expressed preference for a given candidate.
+   * The second preference for Alice and third preference for Bob are removed, leaving a vote
+   * that is invalid for other reasons - the 4th preference is then dropped.
+   * @throws IRVParsingException never
+   */
+  @Test
+  public void duplicateCandidatesTest3() throws IRVParsingException {
+    testUtils.log(LOGGER,"duplicateCandidatesTest3");
+
+    IRVChoices b = new IRVChoices("Alice(1),Alice(2),Bob(2),Chuan(4),Bob(3)");
+    assertEqualListsOfStrings(List.of("Alice","Bob"), b.getValidIntentAsOrderedList());
+  }
+
+  /**
+   * Apply rule 26.7.3 - removing all but the highest expressed preference for a given candidate.
+   * Only the first expressed preference for Alice is retained.
+   * @throws IRVParsingException never
+   */
+  @Test
+  public void duplicateCandidatesTest4() throws IRVParsingException {
+    testUtils.log(LOGGER,"duplicateCandidatesTest4");
+
+    IRVChoices b = new IRVChoices("Alice(1),Alice(2),Alice(4)");
+    assertEqualListsOfStrings(List.of("Alice"), b.getValidIntentAsOrderedList());
+  }
+
+  /**
+   * Apply rule 26.7.3 - removing all but the highest expressed preference for a given candidate.
+   * Alice(2) is removed, leaving a skipped 2nd rank, so then the later (3rd) rank is also dropped.
+   * @throws IRVParsingException never
+   */
+  @Test
+  public void duplicateCandidatesTest5() throws IRVParsingException {
+    testUtils.log(LOGGER,"duplicateCandidatesTest5");
+
+    IRVChoices b = new IRVChoices("Alice(1),Alice(2),Bob(3)");
+    assertEqualListsOfStrings(List.of("Alice"), b.getValidIntentAsOrderedList());
+  }
+
```

Comment:
I would have thought there was an existing library method for comparing lists of strings; anyway -- should this be in test utils?

Reply:
Yes there was! Now sub'd with CollectionUtils.isEqualCollection.